src/service/triple-store-service.js

```diff
@@ -26,462 +32,216 @@ class TripleStoreService {
-            this.repositoryImplementations[repository],
-            repository,
+        const knowledgeCollectionAnnotations = this.dataService.createTripleAnnotations(
+            knowledgeCollection,
+            UAL_PREDICATE,
             ual,
-            assertionId,
         );
-    }
+        const knowledgeCollectionWithAnnotations = [
+            ...knowledgeCollection,
+            ...knowledgeCollectionAnnotations,
+        ];
 
-    async updateAssetAssertionLink(
-        repository,
-        blockchain,
-        contract,
-        tokenId,
-        oldAssertionId,
-        newAssertionId,
-    ) {
-        const ual = this.ualService.deriveUAL(blockchain, contract, tokenId);
-        this.logger.info(
-            `Updating Assertion for the Knowledge Asset with the UAL: ${ual} in the Triple Store's ${repository} repository. ` +
-                `Old Assertion ID: ${oldAssertionId}. New Assertion ID: ${newAssertionId}.`,
-        );
-
-        await this.tripleStoreModuleManager.updateAssetAssertionLink(
-            this.repositoryImplementations[repository],
-            repository,
-            ual,
-            oldAssertionId,
-            newAssertionId,
-        );
-
-        const [assetsWithAssertionIdCount] = await this.countAssetsWithAssertionId(
-            repository,
-            oldAssertionId,
-        );
-
-        // Delete old assertion if it was only linked to this Knowledge Asset
-        if (assetsWithAssertionIdCount?.count === 0) {
-            await this.deleteAssertion(repository, oldAssertionId);
-        }
-    }
-
-    async deleteAssetAssertionLink(repository, blockchain, contract, tokenId, assertionId) {
-        const ual = this.ualService.deriveUAL(blockchain, contract, tokenId);
-        this.logger.info(
-            `Deleting Link to the Assertion with the ID: ${assertionId} for the Knowledge Asset with the UAL: ${ual} ` +
-                `from the Triple Store's ${repository} repository.`,
-        );
-
-        await this.tripleStoreModuleManager.deleteAssetAssertionLink(
-            this.repositoryImplementations[repository],
-            repository,
-            ual,
-            assertionId,
-        );
-
-        const [assetsWithAssertionIdCount] = await this.countAssetsWithAssertionId(
-            repository,
-            assertionId,
-        );
-
-        // Delete assertion if it was only linked to this Knowledge Asset
-        if (assetsWithAssertionIdCount?.count === 0) {
-            await this.deleteAssertion(repository, assertionId);
-        }
-    }
-
-    async getAssetAssertionLinks(repository, blockchain, contract, tokenId) {
-        const bindings = await this.tripleStoreModuleManager.getAssetAssertionLinks(
-            this.repositoryImplementations[repository],
-            repository,
-            this.ualService.deriveUAL(blockchain, contract, tokenId),
+        await this.tripleStoreModuleManager.insertKnowledgeCollectionIntoUnifiedGraph(
```

Comment:
we can paralelize these two calls with promise.all, right? delete from old graph and move to new, also maybe could add retries here too and do a rollback if it's failing?